test/tictactoe/game_runner_test.clj

```diff
@@ -0,0 +1,106 @@
+(deftest validation-does-not-loop-when-valid
+  (testing "It does not loop when there is valid input received"
+    (with-out-str "Invalid. Please pick an available option."
+      (is (= "x" (with-in-str "x" (game-runner/input-validation-loop "X" game-runner/markers)))))))
+
+(deftest loops-when-invalid-choice
+  (testing "Loops until valid input is received"
+    (is (= "Invalid. Please pick an available option.\n"
+      (with-out-str "Invalid. Please pick an available option."
+        (with-in-str "1" (game-runner/input-validation-loop "3" game-runner/game-types)))))))
+
+(deftest input-validation-doesnt-loop-when-valid
+  (testing "It does not loop when valid input is given"
+    (is (= "1" (with-in-str "1" (game-runner/input-validation-loop "1" game-runner/game-types))))))
+
+(deftest get-game-type-from-user 
+  (testing "Returns input as keyword" 
+    (with-out-str "test" (game-runner/print-message (game-runner/game-type-message))
+      (is (= "1" (with-in-str "1" (game-runner/pick game-runner/game-types (game-runner/game-type-message))))))))
+
+(deftest get-board-size-from-user
+  (testing "Returns input as keyword when valid choice"
+    (with-out-str "test" (game-runner/print-message (game-runner/board-type-message))
+      (is (= "1" (with-in-str "1" (game-runner/pick game-runner/board-sizes (game-runner/board-type-message))))))))
+
+(deftest get-4x4-board-size-from-user
+  (testing "Get second board option from user; returns choice as keyword"
+    (with-out-str "test" (game-runner/print-message (game-runner/board-type-message))
+      (is (= "2" (with-in-str "2" (game-runner/pick game-runner/board-sizes (game-runner/board-type-message))))))))
+
+(deftest get-player-one-marker
+  (testing "Get first players marker and return it"
+    (with-out-str "test" (game-runner/print-message (game-runner/player-marker-message "player 1"))
+      (is (= "X" (with-in-str "x" (game-runner/pick-marker)))))))
+
+(deftest get-player-two-marker
+  (testing "Get second players marker and return it"
+   (with-out-str "test" (game-runner/print-message (game-runner/player-marker-message "player 2"))
+     (is (= "O" (with-in-str "o" (game-runner/pick-marker "X")))))))
+
+(deftest show-chosen-game-options
+  (testing "Prints out game options chosen by user"
+   (with-out-str "test" (game-runner/print-message (game-runner/game-choices-message))
+     (is (= "You've chosen:\ngame => 1\nboard => 2\nplayer1 => X\nplayer2 => O\n"
+       (with-out-str (game-runner/chosen-game-options {:game 1 :board 2 :player1 "X" :player2"O"})))))))
+
+(deftest show-game-menu
+  (testing "Gets all game options from user and returns hash of options"
+    (with-out-str "test" (with-in-str (make-input '("2" "2" "a" "b")) 
+      (game-runner/menu game-runner/game-types game-runner/board-sizes))))
+        (is (= {:game "1" :board "1" :player1 "A" :player2 "B"})))
+
+(deftest test-start 
+  (with-out-str "test" (with-in-str (make-input '("1" "2" "a" "b")) (game-runner/start))))
```

Comment:
When I'm reading these tests all I can really see is `with-out-str` and `with-in-str`. Is there a way to refactor so the tests more clearly reveal what exactly they're testing?